js/views/modals/editListing/EditListing.js

```diff
@@ -1437,10 +1482,35 @@ export default class extends BaseModal {
         if (this.cryptoCurrencyType) this.cryptoCurrencyType.remove();
         this.cryptoCurrencyType = this.createChild(CryptoCurrencyType, {
           model: this.model,
+          getCoinTypes: this.getCoinTypesDeferred.promise(),
+          getReceiveCur: () => this._receiveCryptoCur,
         });
         this.getCachedEl('.js-cryptoTypeWrap')
           .html(this.cryptoCurrencyType.render().el);
 
+        const activeCurs = this.model.get('metadata')
+          .get('acceptedCurrencies');
+        let curSelectorInitialState = {
+          currencies: [
+            ...activeCurs,
+            ...supportedWalletCurs(),
+          ],
+          activeCurs,
+          sort: true,
+        };
+        if (this.cryptoCurSelector) {
+          this.cryptoCurSelector.remove();
+          curSelectorInitialState = {
+            ...curSelectorInitialState,
+            ...this.cryptoCurSelector.getState(),
```

Comment:
Does this work after it's removed?

Reply:
It should. Removal is just dom cleanup mainly and I'm not aware of any of our code clearing state on remove. But, I could re-arrange to make it less confusing.